contracts/l1/L1CrossDomainEnabled.sol

```diff
@@ -15,39 +15,31 @@
 
 pragma solidity ^0.6.11;
 
+import "../arbitrum/IBridge.sol";
 import "../arbitrum/IInbox.sol";
 import "../arbitrum/IOutbox.sol";
 
 abstract contract L1CrossDomainEnabled {
-  address public immutable inbox;
+  IInbox public immutable inbox;
 
   event TxToL2(address indexed from, address indexed to, uint256 indexed seqNum, bytes data);
 
   constructor(address _inbox) public {
-    inbox = _inbox;
+    inbox = IInbox(_inbox);
   }
 
   modifier onlyL2Counterpart(address l2Counterpart) {
-    address _inbox = inbox;
-
     // a message coming from the counterpart gateway was executed by the bridge
-    address bridge = address(IInbox(_inbox).bridge());
+    address bridge = inbox.bridge();
     require(msg.sender == bridge, "NOT_FROM_BRIDGE");
 
     // and the outbox reports that the L2 address of the sender is the counterpart gateway
-    address l2ToL1Sender = getL2ToL1Sender(_inbox);
+    address l2ToL1Sender = IOutbox(IBridge(bridge).activeOutbox()).l2ToL1Sender();
+    require(l2ToL1Sender != address(0), "NO_SENDER");
```

Comment:
can we drop this require? i think it's fine but i will run it through arbitrum team later